### PR TITLE
fix(device): capabilities

### PR DESCRIPTION
**Pull request by rokam**
## Summary by CodeRabbit

- **Bug Fixes**
  - Improved error handling in device capability checks to prevent unnecessary exceptions.
  - Corrected the data type for `DeviceAttributes.MODES` to enhance compatibility.

### CODE REVIEW

**Review by chemelli74 — COMMENTED** [review]
Didn't get time to test, but looks good

**Review by chemelli74 — APPROVED** [review]

**Review by wuwentao — APPROVED** [review]

File: midealocal/devices/ac/__init__.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def __init__(
                 DeviceAttributes.fresh_air_mode: None,
                 DeviceAttributes.fresh_air_1: None,
                 DeviceAttributes.fresh_air_2: None,
-                DeviceAttributes.MODES: dict[str, bool],
+                DeviceAttributes.MODES: {},
```
Comment (chemelli74) [inline midealocal/devices/ac/__init__.py:136]: We should always type all vars.

```python
DeviceAttributes.MODES: dict[str, bool] = {}
```

is a better approach

**Comment by sanderlv** [comment]
Sorry, 0.5.1 still hass issue

Logger: midealocal.device
Source: /usr/local/lib/python3.12/site-packages/midealocal/device.py:239
First occurred: 18:26:29 (1 occurrences)
Last logged: 18:26:29

[146235046613512] Unknown error : /usr/local/lib/python3.12/site-packages/midealocal/device.py, 219
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/midealocal/device.py", line 219, in connect
    self.refresh_status(wait_response=True)
  File "/usr/local/lib/python3.12/site-packages/midealocal/device.py", line 344, in refresh_status
    raise SocketException
midealocal.exceptions.SocketException

**Comment by chemelli74** [comment]
This is a PR and not a place to discuss issues.

**Pull request merged**